### DISCUSSION TITLE
Cannot use operator[] with a string argument with string

### DISCUSSION THREAD

**Discussion by kevinzezel** [nlohmann/json#3395] [Q&A] [answered]
Hi,

I'm trying to convert a string to json, which I got from an http request. I got this error:

response string: "{\"status\": 0, \"obs\": \"pong\"}"

Code:
```
http::Request request_get2{"http://localhost:8000/mqtt_ping"};
const auto response_get2 = request_get2.send("GET");
std::string res2 = {response_get2.body.begin(), response_get2.body.end()};
json test  = json::parse(res2);
std::cout << test << std::endl;
std::cout << test["status"] << std::endl;
```

Error:
```
"{\"status\": 0, \"obs\": \"pong\"}"

terminate called after throwing an instance of 'nlohmann::detail::type_error'
what():  [json.exception.type_error.305] cannot use operator[] with a string argument with string
```

Regards

**Comment by nlohmann** [comment, marked as answer]
Can you please check the output of `test.type_name()`? It seems it is a string (hence the exception) rather than an object. Can you also check whether the first byte of `res2` is a quote `"` or a brace `{`?

**Reply by kevinzezel** [reply]
Thanks for the answer!

`std::cout << res2[0] << std::endl; ` is `"`

and 

`std::cout << test.type_name() << std::endl;` is `string`

**Reply by nlohmann** [reply]
Then `res2` is a string and not an object, and hence the exception. Your JSON input seems to be escaped.
